### PR TITLE
Feature: Highlight own Golden/Diamond Goblin

### PR DESCRIPTION
**Pull request by Thunderblade73**
## What
Highlights the own golden/diamond goblin in green.

<details>
<summary>Images</summary>

</details>

## Changelog New Features
+ Added option to highlight own Golden/Diamond Goblin. - Thunderblade73

## Changelog Technical Details
+ Added direct support of setting the color of a mob. - Thunderblade73

### CODE REVIEW

File: src/main/java/at/hannibal2/skyhanni/features/mining/GoldenGoblinHighlight.kt [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    @SubscribeEvent
+    fun onMobEvent(event: MobEvent.Spawn.SkyblockMob) {
+        if (!isEnabled()) return
+        if (!goblinPattern.matches(event.mob.name)) return
+        lastGoblin = event.mob
+        lastGoblinSpawn = SimpleTimeMark.now()
+        handel()
+    }
+
+    private fun handel() {
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/features/mining/GoldenGoblinHighlight.kt]: typo in function name

File: src/main/java/at/hannibal2/skyhanni/features/mining/GoldenGoblinHighlight.kt [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    private var lastGoblinSpawn = SimpleTimeMark.farPast()
+    private var lastGoblin: Mob? = null
+
+    @SubscribeEvent
+    fun onChatEvent(event: LorenzChatEvent) {
+        if (!isEnabled()) return
+        if (!MiningNotifications.goldenGoblinSpawn.matches(event.message) &&
+            !MiningNotifications.diamondGoblinSpawn.matches(event.message)
+        ) return
+        lastChatMessage = SimpleTimeMark.now()
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/features/mining/GoldenGoblinHighlight.kt:35]: set later in the handle function

File: src/main/java/at/hannibal2/skyhanni/features/mining/GoldenGoblinHighlight.kt [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        handel()
+    }
+
+    @SubscribeEvent
+    fun onMobEvent(event: MobEvent.Spawn.SkyblockMob) {
+        if (!isEnabled()) return
+        if (!goblinPattern.matches(event.mob.name)) return
+        lastGoblin = event.mob
+        lastGoblinSpawn = SimpleTimeMark.now()
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/features/mining/GoldenGoblinHighlight.kt:44]: set later in the handle function
x2

**Review by CalMWolfs — CHANGES_REQUESTED** [review]
also seeing as lastChatMessage and lastGoblinSpawn are set at the same point could they just be merged together?

**Comment by Thunderblade73** [comment]
Problem is either chat is faster or the spawn, this way it does not matter which came first. And no they are not set at the same time.

**Comment by CalMWolfs** [comment]
> Problem is either chat is faster or the spawn, this way it does not matter which came first. And no they are not set at the same time.

Oh i read farPast as now 😭

**Review by CalMWolfs — APPROVED** [review]

**Pull request merged**